### PR TITLE
Fix #374, Add Object Utility API functional tests

### LINKED ISSUE #374: Object Utility API's missing functional test
**Is your feature request related to a problem? Please describe.**
OS_IdentifyObject, OS_ConvertToArrayIndex and OS_ForEachObject missing explicit functional tests

**Describe the solution you'd like**
Add tests

**Describe alternatives you've considered**
None.

**Additional context**
Certification violation

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by yammajamma**
Fix #374, Object Utility API's missing functional test

**Describe the contribution**
Created missing functional tests for the OS_IdentifyObject, OS_ConvertToArrayIndex and OS_ForEachObject functions.  

**Testing performed**
Steps taken to test the contribution:
1. Standard build and ran unit tests.

**Expected behavior changes**

**System(s) tested on**
cFS Dev Server
OS: Ubuntu 18.04
Versions: OSAL 5.0.11.0

**Contributor Info **
Yasir Khan
NASA GSFC

### CODE REVIEW

**Comment by skliper** [comment]
> Can we default all text files to be .md instead of .txt?

I'd recommend deleting the .txt file, doesn't add anything.

**Comment by skliper** [comment]
Rebased for bundle CI testing with (mostly) IC branches (broken sample_app commit excluded, ut macro commit included).

Bundle CI passes: https://travis-ci.com/github/skliper/cFS/builds/165460267

**Comment by skliper** [comment]
CCB 20200513 - Discussed, will rebase and review offline (concept good)

**Comment by skliper** [comment]
Rebased and ready for review

**Comment by astrogeco** [comment]
@jphickey can you take a look at this?

File: src/tests/idmap-api-test/idmap-api-test.c [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    UtAssert_True(actual == expected , "OS_ConvertToArrayIndex() (%ld) == %ld ", (long)actual, (long)expected );
+    UtAssert_True(TestArrayIndex == 2 , "TestArrayIndex(%lu) == 2", (long)TestArrayIndex);
+
+    actual   = OS_ConvertToArrayIndex(mutex_id3, &TestArrayIndex);
+    UtAssert_True(actual == expected , "OS_ConvertToArrayIndex() (%ld) == %ld ", (long)actual, (long)expected );
+    UtAssert_True(TestArrayIndex == 3 , "TestArrayIndex(%lu) == 3", (long)TestArrayIndex);
+
+    actual   = OS_ConvertToArrayIndex(time_base_id, &TestArrayIndex);
+    UtAssert_True(actual == expected , "OS_ConvertToArrayIndex() (%ld) == %ld ", (long)actual, (long)expected );
+    UtAssert_True(TestArrayIndex == 1 , "TestArrayIndex(%lu) == 1", (long)TestArrayIndex);
```
Comment (jphickey) [inline src/tests/idmap-api-test/idmap-api-test.c:193]: All of these "OS_ConvertToArrayIndex" tests should _not_ check for specific values -- there should be no assumption that the resources allocated will be in slot "1" .. they could be any value.

Rather - recommend to just check that the output is in range, that is, >=0 and <MAX (where MAX is the configured maximum number of objects for that type).

Comment (jphickey) [reply src/tests/idmap-api-test/idmap-api-test.c:193]: Also, for the cases where more than one resource is allocated, the output array indices should be different from each other (but could still be any value).

Comment (skliper) [reply src/tests/idmap-api-test/idmap-api-test.c:193]: Is there something unique that could be checked to know the right one was returned?

Comment (jphickey) [reply src/tests/idmap-api-test/idmap-api-test.c:193]: Not in the context of a functional test .... This API just outputs a number between 0 and MAX that doesn't overlap with any other resource of the same type currently in use.  So these would be the only two things that a functional test can verify.  To be really thorough, the test could create all the way up to MAX and ensure that all the returned indices were unique/non overlapping.

In practice, it is based on a mask of the LSBs and a MOD operation -- but that is white-box test material, specifically not the type of thing to be tested in a black-box functional test.

Comment (skliper) [reply src/tests/idmap-api-test/idmap-api-test.c:193]: Copy, concur.

**Pull request merged**